backend/api/Services/RobotService.cs

```diff
@@ -224,6 +225,24 @@ public async Task<Robot> UpdateCurrentMissionId(string robotId, string? currentM
 
         public async Task<Robot> UpdateCurrentArea(string robotId, Area? area) { return await UpdateRobotProperty(robotId, "CurrentArea", area); }
 
+        public async Task<Robot> UpdateCurrentInstallation(string robotId, Installation? installation)
+        {
+            var robotQuery = context.Robots.Where(robot => robot.Id == robotId).Include(robot => robot.CurrentInstallation);
+            var robot = await robotQuery.FirstOrDefaultAsync();
+            ThrowIfRobotIsNull(robot, robotId);
+
+            await VerifyThatUserIsAuthorizedToUpdateDataForInstallation(robot!.CurrentInstallation);
+            await VerifyThatUserIsAuthorizedToUpdateDataForInstallation(installation);
+
+            await robotQuery.ExecuteUpdateAsync(robots => robots.SetProperty(r => r.CurrentInstallation, installation));
```

Comment:
Should UpdateCurrentInstallation modify/check currentArea as well so we don't get a mismatch for installation? Or maybe that should be handled elsewhere?

Reply:
CurrentArea should be null unless to robot is localized and driving. Thus changing the installation while the robot is operating isn't something we should be doing. If you want to handle the edge case then one can check if the robot is available or not but I would say this is not necessary.